### ISSUE TITLE
[Question] Why ignoring 'forAllMatchingResources()`?

### ISSUE THREAD

**Issue by Naoray** [tunezilla/nova-dynamic-action-fields#1]
HI there,

just wondering if there is a special case where it would make sense to ignore `ActionRequests` that do have selected the resources by using `Select Matching` (s. [related code](https://github.com/tunezilla/nova-dynamic-action-fields/blob/master/src/DynamicFieldAction.php#L24))?

**Comment by tunezilla** [comment]
Hey, I'm not sure why that check is performed - would need a time machine :sweat_smile: 

If this breaking something for you, do you have an example flow where `?resources=all` is sent to an action? Our use of this package has been for single resources only, so I don't think this is something we run in to.

Thanks!

**Comment by Naoray** [comment]
Thanks for your fast response!

Yes, we have several actions where the user sometimes selects 'all matching' which resulted into missing fields for those actions. We currently added our own `DynamicFieldAction` Trait which uses yours under the hood and overwrites the `fields()` method to get around the described issue.

```php
use Laravel\Nova\Http\Requests\ActionRequest;
use TuneZilla\DynamicActionFields\DynamicFieldAction as BaseDynamicFieldAction;

trait DynamicFieldAction
{
    use BaseDynamicFieldAction;

    public function fields()
    {
        /** @var ActionRequest $actionRequest */
        $actionRequest = $this->actionRequest ?? app(ActionRequest::class);

        $models = [];

        if ($actionRequest->has('resources')) {
            $actionRequest->chunks(100, function ($chunkedModels) use (&$models) {
                /** @var \Illuminate\Database\Eloquent\Collection $chunkedModels */
                $models = array_merge($models, $chunkedModels->all());
            });
        }

        return $this->fieldsForModels(collect($models));
    }
}
```

**Comment by tunezilla** [comment]
Released as 1.0.0:

- https://packagist.org/packages/tunezilla/dynamic-action-fields#v1.0.0
- https://github.com/tunezilla/nova-dynamic-action-fields/releases/tag/v1.0.0

Cheers!

**Comment by Naoray** [comment]
Thanks a lot @tunezilla!
